### DISCUSSION TITLE
redgifs [request]

### DISCUSSION THREAD

**Discussion by theaulddubliner** [tophf/mpiv#47] [MPIV host rules] [answered]
Sample thumb URL: https://redgifs.com/watch/unrulyjauntyinsect

It's the adult version of gfycat.

**Comment by darkred** [comment, marked as answer]
```
{"r": "redgifs\\.com/watch/\\w+$",
 "q": ".video.media > source:nth-child(2)"
}
```
Note that it won't work on github page here because of the site's CSP.

**Reply by tophf** [reply]
`xhr` won't help because github's CSP doesn't allow neither `data` nor `blob` URLs. There's nothing MPIV can do about it. You can use an extension to disable CSP on github entirely or to add those types of URLs. BTW there should be no need for js code in `q`, just use `".video.media > source:nth-child(2)"`

**Reply by darkred** [reply]
Thanks for the info and the improvement.

So, I just added this to the wiki repo:

```json
{"r": "redgifs\\.com/watch/\\w+$",
 "q": ".video.media > source:nth-child(2)"
}
```

**Reply by pintassilgo** [reply]
This works for me even on GitHub:

```js
{
 "r": "redgifs\\.com/watch/\\w+$",
 "q": "return doc.querySelector('meta[property=\"og:video\"]').getAttribute('content')"
}
```

I didn't change nothing about CSP.

**Reply by tophf** [reply]
@pintassilgo, you probably use Firefox + Greasemonkey/Firemonkey which isolate userscripts, as well as Violentmonkey on sites with strict CSP.
